src/main/java/frc/robot/autos/AutoRoutines.java

```diff
@@ -385,4 +387,26 @@ private static final Optional<Rotation2d> turnToNoteOverride() {
     }
     return Optional.of(Rotation2d.fromDegrees(rot));
   }
+
+  /**
+   * Checks if a given path is a midline path (defined as ending at the midline)
+   * This works with our current naming convnetion (having the target note as the last number)
+   * 
+   * @param pathName - String name of a path
+   * @return true if it's a path that ends at the midline, otherwise false
+   */
+  private static boolean isMidlineAuto(String pathName){
+    if (!pathName.contains("return")) {
+      for (int i = pathName.length(); i > 1; i--) {
+          try {
+              if (Integer.parseInt(pathName.substring(i - 1, i)) >= 4) {
```

Comment:
It's a good idea but it's expensive to do. For cases non-midline names (notes 1 to 3), this will traverse the entire string (not the expensive part) but converting to int (parseInt) is expensive.

One way to optimize is to use Character.isDigit(), something like:

      for (int i = pathName.length(); i >1; i--) {
            char ch = input.charAt(i);
            if (Character.isDigit(ch) ) {
              int noteNumber = ch - '0';
               return noteNumber >= 4;
            }
        }
       return false;

Basically, it will stop at the first number it finds from the back of the string and returns true only if the number is >=4